torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -80,10 +80,11 @@ namespace {
 
 void backward(
     const Tensor& self,
-    const Tensor& gradient,
+    c10::optional<Tensor> gradient,
     c10::optional<bool> keep_graph,
     bool create_graph) {
-  torch::autograd::backward({self}, {gradient}, std::move(keep_graph), create_graph);
+  Tensor _gradient = gradient.has_value() ? *gradient : Tensor();
+  torch::autograd::backward({self}, {_gradient}, std::move(keep_graph), create_graph);
```

Comment:
Add a TODO to make `torch::autograd::backward` directly take `c10::optional` here?